protocols/kad/src/behaviour.rs

```diff
@@ -2551,6 +2583,7 @@ where
             }
             FromSwarm::DialFailure(dial_failure) => self.on_dial_failure(dial_failure),
             FromSwarm::AddressChange(address_change) => self.on_address_change(address_change),
+            FromSwarm::NewExternalAddrOfPeer(addr) => self.on_new_address_event(addr),
```

Comment:
Wouldn't it be simpler to just delegating to `Kademlia::add_address` here?

Reply:
yep :)